src/presentation/ui/page/mappings.templ

```diff
@@ -0,0 +1,338 @@
+							)
+						}
+					</div>
+				</div>
+			}
+		</div>
+	</table>
+}
+
+templ VirtualHostRegularForm() {
+	<form
+		hx-post="/api/v1/vhosts/"
+		hx-indicator="#loading-overlay"
+		hx-swap="none"
+	>
+		<div class="my-6">
+			@componentForm.SelectInput("type", "Type", "virtualHost.type", valueObject.AvailableVirtualHostsTypes, false, false)
+		</div>
+		<template x-if="virtualHost.type == 'top-level'">
+			<div class="mb-6">
+				@componentForm.InputField("text", "hostname", "Hostname", "virtualHost.hostname", false)
+			</div>
+		</template>
+		<template x-if="virtualHost.type == 'subdomain'">
+			<div class="mb-6 space-y-6">
+				@componentForm.InputField("text", "hostname", "Subdomain", "virtualHost.hostname", false)
+				@componentForm.InputField("text", "parentHostname", "Parent Hostname", "virtualHost.parentHostname", false)
+			</div>
+		</template>
+		<template x-if="virtualHost.type == 'alias'">
+			<div class="mb-6 space-y-6">
+				@componentForm.InputField("text", "hostname", "Alias", "virtualHost.hostname", false)
+				@componentForm.InputField("text", "parentHostname", "Parent Hostname", "virtualHost.parentHostname", false)
+			</div>
+		</template>
+		<template x-if="virtualHost.hostname == ''">
+			@componentForm.SubmitButton("Create", "ph-check-fat", "closeCreateVhostModal()", false, true)
+		</template>
+		<template x-if="virtualHost.hostname != ''">
+			@componentForm.SubmitButton("Create", "ph-check-fat", "closeCreateVhostModal()", false, false)
+		</template>
+	</form>
+}
+
+templ CreateVhostModal() {
+	@componentStructural.Modal(
+		"Create Virtual Host", "isCreateVhostModalOpen", "closeCreateVhostModal()",
+		"",
+	) {
+		@VirtualHostRegularForm()
+	}
+}
+
+func filterOnlyVhostsHostnames(vhostsWithMappings []dto.VirtualHostWithMappings) []string {
+	vhostsHostnames := []string{}
+	for _, vhostWithMapping := range vhostsWithMappings {
+		vhostsHostnames = append(vhostsHostnames, vhostWithMapping.Hostname.String())
+	}
+
+	return vhostsHostnames
+}
+
+func getOnlyServiceNames() []string {
+	onlyServiceNames := []string{}
+	for serviceName := range valueObject.NativeSvcNamesWithAliases {
+		onlyServiceNames = append(onlyServiceNames, serviceName)
+	}
+
+	return onlyServiceNames
+}
+
+templ MappingRegularForm(vhostsWithMappings []dto.VirtualHostWithMappings) {
+	<form
+		hx-post="/api/v1/vhosts/mapping/"
+		hx-indicator="#loading-overlay"
+		hx-swap="none"
+	>
+		<div class="-mt-4 my-6 space-y-6">
+			@componentMappings.MappingTypeSelectInput("targetType", "Type", "mapping.targetType", "clearMappingTargetValue()")
+			@componentMappings.MappingTypeInfoDialog()
+			<template x-if="isCreateMappingFromVhost">
+				@componentForm.ReadOnlyField("text", "hostname", "Hostname", "virtualHost.hostname", false)
+			</template>
+			<template x-if="!isCreateMappingFromVhost">
+				@componentForm.SelectInput("hostname", "Hostname", "virtualHost.hostname", filterOnlyVhostsHostnames(vhostsWithMappings), true, false)
```

Comment:
Did you pass the entire dto slice just so you could get hostnames? C'mmon man, "Alpine: I don't exist". That is definitely something that should be client-side, you didn't understand what is Alpine responsibility and what is Go responsibility.

Populating a dynamic field is for sure a client-side thing, specially something that will only be loaded AFTER an action for a client. Why would you bother Go to render something that will only happen when a client interacts with the page?

Go is for creating the raw HTML, things that aren't reactive, the base. HTMX is used to updated parts of the HTML that changes after an action and is rendered by Go, like the tables.

If you need to update something that relies on another state, that's why Alpine is here, to manage state and update the page depending on the state. Got it?

Reply:
I moved the vhost hostnames logic to **Alpine**, which was the only reason for this entire workaround.